packages/block-library/src/media-text/media-container.js

```diff
@@ -45,24 +45,16 @@ class MediaContainer extends Component {
 	}
 
 	renderToolbarEditButton() {
-		const { mediaId, onSelectMedia } = this.props;
+		const { onSelectMedia, onSelectURL, mediaUrl } = this.props;
 		return (
 			<BlockControls>
-				<ToolbarGroup>
-					<MediaUpload
-						onSelect={ onSelectMedia }
-						allowedTypes={ ALLOWED_MEDIA_TYPES }
-						value={ mediaId }
-						render={ ( { open } ) => (
-							<IconButton
-								className="components-toolbar__control"
-								label={ __( 'Edit media' ) }
-								icon="edit"
-								onClick={ open }
-							/>
-						) }
-					/>
-				</ToolbarGroup>
+				<MediaReplaceFlow
+					mediaURL={ mediaUrl }
+					allowedTypes={ ALLOWED_MEDIA_TYPES }
+					accept="image/*"
```

Comment:
We also need to be able to upload video files.

Reply:
howdy @Soean fixed this!